sklearn/tests/test_min_dependencies_readme.py

```diff
@@ -0,0 +1,40 @@
+"""Tests for the minimum dependencies in the README.rst file."""
+
+import os
+import re
+from packaging.version import parse
+
+import pytest
+import sklearn
+from sklearn._build_utils.min_dependencies import dependent_packages
+
+
+def test_min_dependencies_readme():
+    # Test that the minimum dependencies in the README.rst file are
+    # consistent with the minimum dependencies defined at the file:
+    # sklearn/_build_utils/min_dependencies.py
+
+    pattern = re.compile(r"(\.\. \|)" +
+                         r"(([A-Za-z]+\-?)+)" +
+                         r"(MinVersion\| replace::)" +
+                         r"( [0-9]+\.[0-9]+(\.[0-9]+)?)")
+
+    readme_path = os.path.abspath(os.path.join(sklearn.__path__[0], ".."))
+    readme_filename = os.path.join(readme_path, "README.rst")
+
+    if not os.path.exists(readme_filename):
+        # Skip because the README.rst
+        # file is not found in Windows
+        pytest.skip("README.rst is not available.")
```

Comment:
This is strange. The windows build script builds the wheel and installs the wheel for testing. The wheel that should have the `README.rst` since it is a part of the `MANIFEST.in`.

Reply:
It is quite strange. I have tried several solutions (`os.path.abspath` and `pathlib.Path`) to use the semantics appropriate for different operating systems and none of them work.

For instance, using [`pathlib.Path`](https://github.com/scikit-learn/scikit-learn/commit/389995d80bf3f57cc251dca64954a15eba3b450a), [this workflow fails](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=21779&view=logs&jobId=d32b16b6-cb9d-571b-e765-de83708fb8dd&j=d32b16b6-cb9d-571b-e765-de83708fb8dd&t=b93f76c1-c2c9-579e-c2ec-c4f438af1261) although:

* `sklearn` is in `c:\miniconda\envs\testvenv\lib\site-packages\sklearn`
* `README.rst` is in `c:\\miniconda\\envs\\testvenv\\lib\\site-packages\\README.rst`

and they are located in the same (parent) directory. Maybe it is related with the path separator (`\` and `\\`), but they are handled by `pathlib.Path`.

On the other hand, using  [`os.path.abspath`](https://github.com/scikit-learn/scikit-learn/pull/18301/commits/a4b66baea8569c79f77a9d793215a2a3bf8b2dde), [this workflow fails](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=21772&view=logs&jobId=d32b16b6-cb9d-571b-e765-de83708fb8dd&j=d32b16b6-cb9d-571b-e765-de83708fb8dd&t=b93f76c1-c2c9-579e-c2ec-c4f438af1261) for the same reason.

Any ideas?

Reply:
Maybe using [pathlib](https://docs.python.org/3/library/pathlib.html) would work?

I have always found it nicer to use and it is cross platform.

Reply:
Indeed, that was my first approach, but it does not work. Nevertheless, I have changed `os.path.abspath` by `pathlib.Path` because I prefer this version too.